test/libsolidity/syntaxTests/code/usage_as_identifier/usage_as_int_variable.sol

```diff
@@ -0,0 +1,8 @@
+contract X {
+    function codeAsString() external {
+        uint32 calldata code;
+    }
+}
+// ----
+// ParserError 3548: (76-80): Location already specified.
```

Comment:
Once this gets released, I would expect something like this to be the most frequent misuse of the `code` keyword, and this error (or rather the combination of them) is not as descriptive as I would like. Sure, if you know about the new `code` location, it's easy to decipher, but if you don't, it's a little bit too cryptic.

Now, C++ and Java use our current approach, where there's no mention of a reserved keyword, but Rust for instance produces a very nice error, e.g.
```rust
fn main() {
    let break = 10;
}
```
will produce the following output when you attempt to compile
```
error: expected identifier, found keyword `break`
 --> src/main.rs:2:9
  |
2 |     let break = 10;
  |         ^^^^^ expected identifier, found keyword
  |
```
My point here is a bit of digression, and is not really in the scope of this PR, but just felt like mentioning it.